src/ansys/dpf/core/animator.py

```diff
@@ -152,6 +152,130 @@ def animation():
+                    if "imageio ffmpeg plugin you need" in e.msg:
+                        raise ImportError(
+                            "The imageio-ffmpeg library is required to save "
+                            "animations. Please install it first with the command "
+                            "'pip install imageio-ffmpeg'"
+                        )
+                    else:
+                        raise e
+        freq_kwargs = kwargs.pop("freq_kwargs", {})
+
+        # Camera position
+        cpos = kwargs.pop("cpos", None)
+        if cpos:
+            if isinstance(cpos[0][0], float):
+                cpos = [cpos] * len(indices)
+
+        workflow.connect(input_name, [mode_number])
+
+        initial_field = workflow.get_output(output_name, core.types.field)
+        max_ = float(np.max(initial_field.data))
+
+        def render_frame(frame):
+            self._plotter.clear()
+
+            field = workflow.get_output(output_name, core.types.field)
+
+            # TODO: deform_by is not accepted for niw
+            # deform = None
+            # if "deform_by" in workflow.output_names:
+            #     deform = workflow.get_output("deform_by", core.types.field)
+
+            ponderation = float(indices[frame])
+            op = core.operators.math.scale(
+                field=field,
+                ponderation=ponderation,
+            )
+            field_ponderated = op.outputs.field()
+
+            clim = [0, max_]
+
+            self.add_field(
+                field_ponderated,
+                # deform_by=deform,
+                clim=clim,
+                **kwargs,
+            )
+            kwargs_in = _sort_supported_kwargs(bound_method=self._plotter.add_text, **freq_kwargs)
+            # print(type(f"Mode number {mode_number}"))
+            self._plotter.add_text(f"Mode number {mode_number}\nt={float(frame)}s", **kwargs_in)
+            if cpos:
+                self._plotter.camera_position = cpos[frame]
+
+        try:
+
+            def animation():
+                if save_as:
+                    try:
+                        self._plotter.write_frame()
+                    except AttributeError as e:  # pragma: no cover
+                        if (
+                            "To retrieve an image after the render window has been closed"
+                            in e.args[0]
+                        ):
+                            print("Animation canceled.")
+                            print(e)
+                            return result
+                # For each additional frame requested
+                if len(indices) > 1:
+                    for frame in range(1, len(indices)):
+                        try:
+                            render_frame(frame)
+                        except AttributeError as e:  # pragma: no cover
+                            if "'NoneType' object has no attribute 'interactor'" in e.args[0]:
+                                print("Animation canceled.")
```

Comment:
@PProfizi Do you have a logger for dpf-core to output this content?

Reply:
Hi @MaxJPRey, at the moment we do not have a logger logic, no.